zypp/VendorAttr.cc

```diff
@@ -58,6 +58,24 @@ namespace zypp
     bool equivalent( IdString lVendor, IdString rVendor ) const
     { return lVendor == rVendor || vendorMatchId( lVendor ) == vendorMatchId( rVendor ); }
 
+    unsigned foreachVendorList( std::function<bool(VendorList)> fnc_r ) const
+    {
+      std::map<unsigned,VendorList> lists;
+      for( const auto & el : _vendorGroupMap )
+	lists[el.second].push_back( el.first );
+
+      unsigned ret = 0;
+      for ( auto el : lists ) {
+	VendorList & vlist { el.second };
+	if ( vlist.empty() )
+	  continue;
+	++ret;
+	if ( fnc_r && !fnc_r( std::move(vlist) ) )
```

Comment:
Do you really want to move the vlist out of _vendorGroupMap? Since its a writeable reference into the map that would kill the data there..

Reply:
I move out of lists, not out of _vendorGroupMap.

Reply:
Right, having the variable "el" twice confused me